### ISSUE TITLE
[Artifactory] Conan unable to find a package?

### ISSUE THREAD

**Issue by mpdelbuono** [conan-io/conan#1786]
Here's a weird one, and unfortunately because it looks like the Conan API endpoints are in Artifactory PRO, I don't have source code to try to track this one down.

I have a single package that cannot be found in Artifactory by Conan. I can clearly see the package present in Artifactory and it even mentions that it's a Conan package in the UI. However, when actually trying to use it as a remote in Conan, I start to have trouble. Conan is trying to find this package due to a requirement being set on "Foo/[0.1.x]@autobuilder/main" in the recipe.

Using CONAN_TRACE_FILE I tracked down the issue to the REST API endpoint at /artifactory/api/conan/<repository>/v1/conans/search. The requirement is of the form "Foo/[0.1.x]@autobuilder/main". The REST call ends up with a parameter q of "Foo/*@autobuilder/main". 

I have confirmed manually that the REST call responds with an empty array (i.e., nothing found). However, if I change the REST parameter q to look for "Foo/0.1.0@autobuilder/main" (i.e., replace the wildcard with an explicit version) then the REST query responds with a result (the expected package).

What I don't understand is what is different about this package from any other. I'm using a common template for all of my conanfile.py recipes, so they should all be extremely similar. Looking through the properties and permissions on Artifactory I don't find anything different about this one. The package name is not anything special (it's just letters).

FWIW, I can use other commands to generate the same strange behavior:
```
D:\>conan search -r artifactory Foo/*@autobuilder/main
There are no packages matching the Foo/*@autobuilder/main pattern

D:\>conan search -r artifactory Foo/0.1.0@autobuilder/main
Existing packages for recipe Foo/0.1.0@autobuilder/main:

    Package_ID: d717561b63ab2c34dd3704033751a33c21595e1c
        [options]
            qualified: False
            releaseType: IW
        [settings]
            arch: x86
            build_type: Debug
            compiler: Visual Studio
            compiler.runtime: MDd
            compiler.version: 9
            os: Windows
        outdated from recipe: False
```

Yet, other packages work fine:
```
D:\>conan search -r artifactory CommonLib/*@autobuilder/main
Existing package recipes:

CommonLib/0.1.0@autobuilder/main
```

The only workaround I've found is to change the requirement for `Foo/[0.1.x]@autobuilder/main` to explicitly request `Foo/0.1.0@autobuilder/main` in the recipe, but this is not a desired solution.

I realize this gives you very little information to go off of, but I'm not really sure where to continue the investigation. I imagine artifactory must be failing to find something along the way when it does the search for Foo/* per the REST query, but I don't know how to investigate inside there.

**Comment by mpdelbuono** [comment]
Note that I consider this to be distinct from #1779 because while #1779 is referencing a `conan upload` command, this issue is actually conan itself generating the REST call due to a requirement specified in the recipe (that requirement being `Foo/[0.1.x]@autobuilder/main`, in this case).

**Comment by memsharded** [comment]
What is the name of Foo? Maybe some special character in it? There was a bug for packages containing the sequence of letters "and" in its name, but I think it should be fixed now.

Please also specify your Artifactory and conan versions.

**Comment by mpdelbuono** [comment]
Sorry, I didn't think that the exact name would matter that much beyond "it's just letters", but you're right, it does have "and" in it. The exact package name is "Bandit" and so the requirement is `Bandit/[0.1.x]@autobuilder/main`

Artifactory version: 5.4.6
Conan client version: 0.26.0

I did notice when I was accessing the REST API directly that Artifactory is responding with the following headers. I'm not sure if this is relevant. This appears to be as packaged with Artifactory's installer:
```
X-Conan-Server-Capabilities:
X-Conan-Server-Version: 0.20.0
```

**Comment by mpdelbuono** [comment]
With that extra information, looks like this is a duplicate of #1352 which also mentions that this is fixed in the unreleased Artifactory 5.5.

**Comment by memsharded** [comment]
Yes, exactly that. It was a very unfortunate bug because of a combination of things. You have two workarounds until Artifactory 5.5 is released:
- Just change the name of the package: ``Bndit``, for example
- Use exact versions instead of version ranges for a while.

Regarding version ranges I would like to recall that they might be a bit slow, specially if your repositories are very big. Version ranges require doing a "search" call to the server, to get the possible matching versions, then compute the best match in the range. But a search is an extra call which might be slow. Alternatives you could consider are ``conan alias`` to define latest version at package creation time, or using dependency overriding, and defining latest versions in the consumers. Please have this in mind if you get to this problem.

Closing this as duplicate, please follow #1352 for updates on this issue. Thanks very much for all your feedback and help!
